src/dialogs/config-flow/dialog-data-entry-flow.ts

```diff
@@ -84,59 +91,89 @@ class DataEntryFlowDialog extends LitElement {
     this._params = params;
     this._instance = instance++;
 
-    // Create a new config flow. Show picker
-    if (!params.continueFlowId && !params.startFlowHandler) {
-      if (!params.flowConfig.getFlowHandlers) {
-        throw new Error("No getFlowHandlers defined in flow config");
+    if (params.startFlowHandler) {
+      this._checkFlowsInProgress(params.startFlowHandler);
+      return;
+    }
+
+    if (params.continueFlowId) {
+      this._loading = true;
+      const curInstance = this._instance;
+      let step: DataEntryFlowStep;
+      try {
+        step = await params.flowConfig.fetchFlow(
+          this.hass,
+          params.continueFlowId
+        );
+      } catch (err) {
+        this._step = undefined;
+        this._params = undefined;
+        showAlertDialog(this, {
+          title: "Error",
+          text: "Config flow could not be loaded",
+        });
+        return;
       }
-      this._step = null;
 
-      // We only load the handlers once
-      if (this._handlers === undefined) {
-        this._loading = true;
-        try {
-          this._handlers = await params.flowConfig.getFlowHandlers(this.hass);
-        } finally {
-          this._loading = false;
-        }
+      // Happens if second showDialog called
+      if (curInstance !== this._instance) {
+        return;
       }
-      await this.updateComplete;
-      return;
-    }
 
-    this._loading = true;
-    const curInstance = this._instance;
-    let step: DataEntryFlowStep;
-    try {
-      step = await (params.continueFlowId
-        ? params.flowConfig.fetchFlow(this.hass, params.continueFlowId)
-        : params.flowConfig.createFlow(this.hass, params.startFlowHandler!));
-    } catch (err) {
-      this._step = undefined;
-      this._params = undefined;
-      showAlertDialog(this, {
-        title: "Error",
-        text: "Config flow could not be loaded",
-      });
+      this._processStep(step);
+      this._loading = false;
       return;
     }
 
-    // Happens if second showDialog called
-    if (curInstance !== this._instance) {
-      return;
+    // Create a new config flow. Show picker
+    if (!params.flowConfig.getFlowHandlers) {
+      throw new Error("No getFlowHandlers defined in flow config");
```

Comment:
Why throw an Error and not showAlertDialog?

Reply:
This is not a user error